### ISSUE TITLE
Love the project

### ISSUE THREAD

**Issue by ayroblu** [vifm/vifm#541]
Hey, I love vifm, I think it's great and it works really well, thank you so much!

I have some ideas that I think have been asked before but I'm not clear where it stands. I think the project is great without these ideas, but just would make it basically feature complete for me (so feel free to ignore me / track in other tickets)

Context: I use **netrw** and **fzf** in my daily workflow for file management + **terminal** actions

Killer features I would love to have:
1. show symlinks (netrw shows the path, so it would emulate that)
2. git integration (hide / dull out hidden files) (fzf can be customised to only search on git recognised files)
3. tree mode (netrw allows opening and closing folders without changing directory)

This is one of my favourite projects: https://github.com/jpwilliams/gitree#readme
But I hardly use it as I don't really need an overview. But if I could have gitree + vifm, I'd be super happy as I would have better context while I manage files.

Anyways, just some thoughts. If some of this is already possible with config please let me know!

**Comment by xaizek** [comment]
> Hey, I love vifm, I think it's great and it works really well, thank you so much!

Hello. Nice to hear that!

> 1. show symlinks (netrw shows the path, so it would emulate that)

You're not the first one to ask for that.  At the moment you can add `-{target}` to the value of your `'viewcolumns'` to see where the link points to.

> 2. git integration (hide / dull out hidden files) (fzf can be customised to only search on git recognised files)

By "hidden" you mean untracked or ignored files? You just don't want to see them while browsing a repository?

> 3. tree mode (netrw allows opening and closing folders without changing directory)

You can build a tree out of files tracked by `git`:

    :!git ls-files %u
    :tree

Or all files:

    :tree

The first one is static meaning that it won't be updated to include new files.

There is no collapsing of directories at the moment.

**Comment by ayroblu** [comment]
> > 1. git integration (hide / dull out hidden files) (fzf can be customised to only search on git recognised files)
> 
> By "hidden" you mean untracked or ignored files? You just don't want to see them while browsing a repository?

Yup, perhaps a toggle or something else, but having a gitree like view in vifm would be really cool.
Again, not sure this is super useful, but thought it would be pretty cool at least to hide untracked like IDEs do

> > 1. tree mode (netrw allows opening and closing folders without changing directory)
> 
> You can build a tree out of files tracked by `git`:
> 
> ```
> :!git ls-files %u
> :tree
> ```

Fancy!

> There is no collapsing of directories at the moment.

Yup, normally I care about one or two leaves, especially in larger repos so hope it gets implemented sometime!

**Comment by xaizek** [comment]
> 1. show symlinks (netrw shows the path, so it would emulate that)

If you build from `master`, you can try using `NameLink` view column from [this sample plugin](https://github.com/vifm/vifm/blob/master/data/plugins/viewcolumn/init.lua).

**Comment by xaizek** [comment]
> Yup, normally I care about one or two leaves, especially in larger repos so hope it gets implemented sometime!

Just pushed changes that add <kbd>zx</kbd> key to toggle directory folding, so it will be in the next release.
